adapters/src/main/java/com/squareup/moshi/adapters/PolymorphicJsonAdapterFactory.java

```diff
@@ -220,7 +253,7 @@ private int labelIndex(JsonReader reader) throws IOException {
         return labelIndex;
       }
 
-      throw new JsonDataException("Missing label for " + labelKey);
+      return -1;
```

Comment:
Good point - 64af8bb

Reply:
Why this? I think the default value can be used to return it instead of an Exception if I do not receive the label.

If we have a Polymorphic type in an inner field, we will receive for the full json. Instead, I could use the default value for the uncognized value and return the remaining json which I parsed correctly.